src/app/components/forms/Upload.tsx

```diff
@@ -81,24 +144,82 @@ const Upload = ({
           </Button>
         </span>
       )}
-      {state === "success" && <span>Fichier chargé: {value}</span>}
+      {state === "success" && (
+        <span className="flex items-center justify-center gap-2">
+          <i className="fr-icon-file-text-fill text-action-high-blue-france" />
+          <span className="flex flex-col">
+            <a href={fileData?.fileUrl}>{fileData?.originalName}</a>
+            {fileData?.fileSize && (
+              <span className="text-xs text-default-grey tabular-nums">
+                {prettyBytes(fileData?.fileSize, { locale: "fr" })}
+              </span>
+            )}
+          </span>
+          <Button
+            iconId="fr-icon-delete-bin-line"
+            priority="tertiary no outline"
+            size="small"
+            className="!rounded-full !bg-white"
+            title="Supprimer le fichier"
+            onClick={handleDelete}
+          />
+        </span>
+      )}
+      {state === "error" && (
+        <span className="flex items-center justify-center gap-2">
+          <Button
+            size="small"
+            priority="tertiary no outline"
+            className="bg-white"
+            onClick={handleBrowse}
+            ref={buttonRef}
+          >
+            Parcourir
+          </Button>
+          <p>{errorMessage}</p>
+        </span>
+      )}
+      {state === "deleting" && (
+        <span className="flex items-center justify-center gap-2">
+          Suppression en cours...
+        </span>
+      )}
       <input
-        className="block display-none w-full text-sm text-gray-900 border border-gray-300 rounded-lg cursor-pointer bg-gray-50 dark:text-gray-400 focus:outline-none dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400"
+        className="hidden"
         type="file"
         ref={fileInputRef}
-        onAbort={handleAbort}
         onChange={handleFileChange}
+        accept={accept}
+      />
+      <input
+        type="hidden"
+        onChange={(e) => {
+          onChange?.(e);
+          setValueState(e.target.value);
+        }}
+        name={props.name}
+        value={valueState}
+        {...props}
       />
-      <input type="text" value={value || ""} {...props} />
     </div>
   );
 };
 
 export default Upload;
 
-type UploadStates = "idle" | "loading" | "success" | "error" | "uploaded";
+type UploadStates =
```

Comment:
```suggestion
type UploadState =
```
Pour moi, pas de `s` à la fin puisque ce n'est pas une liste